ckanext/dbquery/actions/dbquery.py

```diff
@@ -35,6 +37,14 @@ def query_database(context, data_dict):
         colnames = []
         message = f"Query affected {result.rowcount} rows"
 
+    # Save executed query
+    if context.get('user'):
+        executed = DBQueryExecuted(
+            query=query,
+            user_id=context.get('user')
+        )
```

Comment:
A user will be always available, there is no need for the `if context.get('user')`. We can remove it
Also, we need to ensure we save the `User` id (not the name)
So, get the `User` object with `user = context['auth_user_obj']` and the use `user_obj.id` 
CKAN provides `auth_user_obj` for this cases. Test it, it must work